### ISSUE TITLE
inaccurate rgb values bug

### ISSUE THREAD

**Issue by TamarWeiss** [photopea/photopea#7261]
I've been experiencing this little bug fore a while, where I've been filling with the bucket tools certain colors, and instead it puts of canvas a color off by 1 (example at the bottom. But recently it seems this bug has sipped into the pencil tool, which was what I used to circumvent this issue.

The values which is seems to deviant from are somewhat consistent (90 and 180 are the ones are remember clearly, but there are more). I don't what might cause it other then some conversion rounding error or what not.

Please fix it. As a sprite artist which is very meticulous with my color choices, this bug is very inconvenient.
Thanks;

![21-10-2024 18-44-25](https://github.com/user-attachments/assets/265cf6cb-7d8c-422e-829d-50d537580a9a)
![21-10-2024 18-44-43](https://github.com/user-attachments/assets/eed60127-8b59-408b-86fb-bb656d040caf)

**Comment by Mayur-hot** [comment]
You are Right, I recorded video for the same issue.. Color RGB Value changed..

[screen-recorder-tue-oct-22-2024-11-32-17.webm](https://github.com/user-attachments/assets/67673773-569e-4321-866e-f43e0ceb7933)

**Comment by Mayur-hot** [comment]
[screen-recorder-tue-oct-22-2024-11-37-56.webm](https://github.com/user-attachments/assets/ed0bbf3e-f6d1-4f81-b84d-e14c712071c0)

**Comment by photopea** [comment]
I am very sorry, it should be fixed now. This bug was introduced this year.

**Comment by TamarWeiss** [comment]
thanks you so much! just for added curiosity - what may had caused that bug? My programmer side is genuinely curious to know.

**Comment by TamarWeiss** [comment]
oh, floating point error. classic.
